### PR TITLE
Add support, test for kwargs in jit._fork (#38357)

### PR DESCRIPTION
**Pull request by wconstab**
Closing 38357

### CODE REVIEW

File: test/jit/test_async.py [hunk 1]
```diff
@@ -197,6 +197,23 @@ def wait_script(x1, x2, x3):
         self.assertEqual(y2, foo2(x1, x2))
         self.assertEqual(y3, foo3(x1, x2, x3))
 
+    def test_async_kwargs(self):
+        @torch.jit.script
+        def foo(x1, x2):
```
Comment (eellison) [inline test/jit/test_async.py:202]: Can you also test with a non-scripted function, and also while using `torch.jit.trace` ?

File: torch/csrc/jit/python/init.cpp [hunk 2]
```diff
@@ -858,7 +858,7 @@ void initJITBindings(PyObject* module) {
         tracer::WithNestedTracingFrame env_guard;
 
         // Run the user-supplied function
-        py_func_output = f(*args_tup);
+        py_func_output = f(*args_tup, **kwargs);
```
Comment (eellison) [inline torch/csrc/jit/python/init.cpp:861]: Don't think this code path is currently being tested

**Review by eellison — COMMENTED** [review]
Looks good! A couple more tests and should be good to go.

File: torch/csrc/jit/python/init.cpp (hunk 2 above)
Comment (wconstab) [reply torch/csrc/jit/python/init.cpp:861]: Yea, you're right.  It's covered with my latest test.

File: test/jit/test_async.py (hunk 1 above)
Comment (wconstab) [reply test/jit/test_async.py:201]: I added different variants.

**Comment by wconstab** [comment]
@eellison There was a failure in nccl tests on one of the CI runs- do you happen to know if that's a flaky test, or something I might have broken?

**Review by eellison — APPROVED** [review]
LGTM.  You can check the state of master at https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master. The tests that's failing on this PR is failing on master as well.

**Pull request closed without merging**